app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -245,6 +225,57 @@ private static boolean shouldGenerateFirstKey() {
         return keyServer == null || keyServer.equals("");
     }
 
+    private ResultAndError<PullTaskResult> getRequestResultOrRetry(AndroidTransactionParserFactory factory) {
+        if (retryLimit != -1 && numTries >= retryLimit) {
+            // for testing purposes only

```

Comment:
Ah. It just wasn't part of the spec or something we discussed introducing at all. I guess in theory we could decide on a limit for number of times that we want to allow us to get retry responses from HQ, but I actually don't think we do, because it's hard to know what how many of those it might take, depending on how huge the restore is and how taxed the server is.

I think it may make more sense once you read through the tests as to why I added this for those only.